tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/method/FirebaseSubscriptionGetMethod.scala

```diff
@@ -28,6 +28,13 @@ class FirebaseSubscriptionGetMethodModule extends AbstractModule {
       .addBinding()
       .to(classOf[FirebaseSubscriptionGetMethod])
   }
+
+  @Provides
+  @Singleton
+  def provideFirebaseSubscriptionGetMethod(repository: FirebaseSubscriptionRepository,
+                                           metricFactory: MetricFactory,
+                                           sessionSupplier: SessionSupplier): FirebaseSubscriptionGetMethod =
+    new FirebaseSubscriptionGetMethod()(repository, metricFactory, sessionSupplier)
 }
```

Comment:
IMO just use normal @Inject and guice binding. The module chooser will take care of the `enable/disable firebase feature`.

Reply:
repository, method... always create instance when runtime even when firebase feature is disable?

Reply:
> The module chooser will take care of the enable/disable firebase feature

That means the module chooser already takes care of whether loading the beans at runtime.